internal/exporter/exporter.go

```diff
@@ -0,0 +1,168 @@
+package exporter
+
+import (
+	"archive/tar"
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"github.com/ethersphere/bee/pkg/shed"
+	"io"
+	"os"
+)
+
+const (
+	// filename in tar archive that holds the information
+	// about exported data format version
+	ExportVersionFilename = ".swarm-export-version"
+	// current export format version
+	CurrentExportVersion = "1"
+	// default export filename
+	DefaultExportFilename = "swarm-exportdb.tar"
+)
+
+type ProgressUpdater interface {
+	Update(int, int)
+}
+
+type Option func(*exporter)
+
+func WithDestinationFilename(fname string) Option {
+	return func(e *exporter) {
+		e.dstFile = fname
+	}
+}
+
+func WithProgressUpdater(upd ProgressUpdater) Option {
+	return func(e *exporter) {
+		e.updater = upd
+	}
+}
+
+func Export(src string, opts ...Option) error {
+	e, err := newExporter(src, opts...)
+	if err != nil {
+		return errors.New("Invalid source directory err:" + err.Error())
+	}
+	return e.export()
+}
+
+type noopUpdater struct{}
+
+func (n noopUpdater) Update(_, _ int) {}
+
+type exporter struct {
+	retrievalIndex shed.Index
+	closer         io.Closer
+	dstFile        string
+	updater        ProgressUpdater
+}
+
+func defaultOpts(e *exporter) {
+	if e.dstFile == "" {
+		e.dstFile = DefaultExportFilename
+	}
+	if e.updater == nil {
+		e.updater = noopUpdater{}
+	}
+}
+
+func getRetrievalIndex(src string) (index shed.Index, closer io.Closer, err error) {
+	s, e := shed.NewDB(src, nil)
+	if e != nil {
+		err = e
+		return
```

Comment:
Just a suggestion, as very subjective clarity improvement.

```suggestion
		return nil, nil, e
```